pkg/cmd/roachtest/test_registry.go

```diff
@@ -87,6 +87,10 @@ func (r *testRegistry) Add(spec testSpec) {
 
 // prepareSpec validates a spec and does minor massaging of its fields.
 func (r *testRegistry) prepareSpec(spec *testSpec) error {
+	if matched, err := regexp.MatchString("^[a-zA-Z0-9-_=/,]+$", spec.Name); err != nil || !matched {
+		return fmt.Errorf("%s: must specify valid Name", spec.Name)
```

Comment:
You could print the regexp here (`regexp.QuoteMeta(pattern)`), that would help those who do come up with rejected test names. As is, they'll have to figure out what a valid name is.